developer-workflow/c-api.rst

```diff
@@ -88,6 +95,40 @@ Guidelines for expanding/changing the public API:
   fields, arguments and return values are well defined.
 
 
+C API Tests
+-----------
+
+Tests for the public C API live in the ``_testcapi`` module.
+Functions named ``test_*`` are used as tests directly.
+Tests that need Python code (or are just easier to partially write in Python)
+live in ``Lib/test``, mainly in :file:`Lib/test/test_capi.py`.
+
+Due to its size, the ``_testcapi`` module is defined in several source
+files.
+To add a new set of tests (or extract a set out of the monolithic
+:file:`Modules/_testcapimodule.c`):
+
+- Create a C named ``Modules/_testcapi/yourfeature.c``
+
+- The file should define a module as usual, except:
+
+  - Instead of ``<Python.h>``, include ``"parts.h"``.
+  - Instead of ``PyInit_modname``, define ``_PyTestCapi_Init_yourfeature``
+    that *takes* the ``_testcapi`` module and adds functions/classes to it.
+    (Use ``PyModule_AddFunctions`` to add functions).
+
+- Add the ``_PyTestCapi_Init_*`` function to ``Modules/_testcapi/parts.h``
+
+- Call the ``_PyTestCapi_Init_*`` from ``PyInit__testcapi`` in
```

Comment:
Is this a typo or is it correct?
```suggestion
- Call the ``_PyTestCapi_Init_*`` from ``PyInit_testcapi`` in
```

Reply:
You have a sharp eye!
But it's [correc](https://github.com/python/cpython/blob/586fc02be5b3e103bfddd49654034a898a8d6dfc/Modules/_testcapimodule.c#L6426)t: [`PyInit_`](https://docs.python.org/3/extending/building.html#c.PyInit_modulename) is prepended to the module name, which itself starts with an underscore.